starter-code/src/viking.js

```diff
@@ -1,11 +1,103 @@
 // Soldier
-class Soldier {}
+class Soldier {
+  constructor (health, strength){
+    this.health = health;
+    this.strength = strength;
+  }
+
+  attack(){
+    return this.strength;
+  }
+
+  receiveDamage(damage){
+    this.health -= damage;
+  }
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+  constructor(name,health,strength){
+    super(health,strength);
+    this.name = name;
+  }
+
+  receiveDamage(damage){
+    this.health -= damage;
+    if (this.health>0){
+      return (this.name+" has received " + damage.toString() + " points of damage");
```

Comment:
Tu peux aussi écrire  return(`${this.name} has received ${damage} points of damage`);